### DISCUSSION TITLE
Weird highlighting on text

### DISCUSSION THREAD

**Discussion by 0x56696B** [LazyVim/LazyVim#4263] [Q&A] [answered]
Hello.
After a recent update of LazyVim, my colorscheme seems to have changed.
I currently get some weird background color on all of the text, everywhere, where as before that background color was not present and the word itself was in that color.
Here is an example:

![image](https://github.com/user-attachments/assets/b8c11fe3-ae7b-4fac-a45d-1bfa19860b9a)

The text itself was supposed to be in the color, not the background and the text being only white/black. 
Also occurs on the dark theme.

Here is my colorscheme file, in case it's needed.
```lua
return {
  {
    'catppuccin/nvim',
    name = 'catppuccin',
    lazy = true,
    priority = 1000,
    config = function()
      -- Actual theme switching occurs way below, at dark-notify (for Mac OS)
      require('catppuccin').setup {
        background = {
          light = "latte",
          dark = 'macchiato',
        },
        color_overrides = {
          -- Dark color palette
          mocha = {
            rosewater = '#ea6962',
            flamingo = '#ea6962',
            red = '#ea6962',
            maroon = '#ea6962',
            pink = '#d3869b',
            mauve = '#d3869b',
            peach = '#e78a4e',
            yellow = '#d8a657',
            green = '#a9b665',
            teal = '#89b482',
            sky = '#89b482',
            sapphire = '#89b482',
            blue = '#7daea3',
            lavender = '#7daea3',
            text = '#ebdbb2',
            subtext1 = '#d5c4a1',
            subtext0 = '#bdae93',
            overlay2 = '#a89984',
            overlay1 = '#928374',
            overlay0 = '#595959',
            surface2 = '#4d4d4d',
            surface1 = '#404040',
            surface0 = '#292929',
            base = '#1d2021',
            mantle = '#191b1c',
            crust = '#141617',
          },
          -- Dark color palette
          macchiato = {
            rosewater = '#eb7a73',
            flamingo = '#eb7a73',
            red = '#eb7a73',
            maroon = '#eb7a73',
            pink = '#e396a4',
            mauve = '#e396a4',
            peach = '#e89a5e',
            yellow = '#e8b267',
            green = '#b9c675',
            teal = '#99c792',
            sky = '#99c792',
            sapphire = '#99c792',
            blue = '#8dbba3',
            lavender = '#8dbba3',
            text = '#f1e4c2',
            subtext1 = '#e5d5b1',
            subtext0 = '#c5bda3',
            overlay2 = '#b8a994',
            overlay1 = '#a39284',
            overlay0 = '#656565',
            surface2 = '#5d5d5d',
            surface1 = '#505050',
            surface0 = '#393939',
            base = '#2e3233',
            mantle = '#242727',
            crust = '#1f2223',
          },
        },
      }
    end,
  },

  {
    "LazyVim/LazyVim",
    opts = {
      colorscheme = "catppuccin",
    },
  },

  {
    'cormacrelf/dark-notify',
    event = 'VeryLazy',
    enabled = vim.fn.has 'macunix',
    priority = 999,
    opts = {
      schemes = {
        dark = 'catppuccin-macchiato',
        light = 'catppuccin-latte',
      },
    },
    config = function(_, opts)
      local dn = require 'dark_notify'
      dn.run(opts)
    end,
  },
}
```

Edit (1):
Changing the colorscheme yielded no result.

Edit (2): 
Disabling `dark-notify` yielded no result.

Edit (3): Rolling back the commits of the config to several months old yielded no result.

Edit (4): Clearing cache and rolling back the commits of the config yielded, again, no result.

**Comment by dpetka2001** [comment]
I don't know what `dark-notify` does but here is a screenshot from default LazyVim catppuccin colorscheme. So, it's probably something with your own configuration.

![2024-08-06_12-49](https://github.com/user-attachments/assets/a3d161cd-2424-467c-8bfd-f6f432ef2498)

**Reply by 0x56696B** [reply]
`dark-notify` is a plugin to dynamically choose theme, based on OS theme for Mac OS.

I couldn't find what I did wrong in my config to get these strange highlightings on the text. 

Even with disabling `dark-notify` and switching the theme to something completely new, such as `gruvbox`, still showed no results:
<img width="561" alt="image" src="https://github.com/user-attachments/assets/4a7dbe33-d9f1-412a-8685-b258c47a19a9">

**Reply by dpetka2001** [reply]
Can you try `:set notermguicolors`?

**Reply by 0x56696B** [reply]
Still there

<img width="1210" alt="image" src="https://github.com/user-attachments/assets/f66777ad-4ad4-4c30-a828-00eaa1c8bec1">

**Reply by dpetka2001** [reply]
Maybe your terminal doesn't support true colors? Try with Kitty or Wezterm. I can't reproduce this and I don't really know how Mac behaves, since I don't have any experience with it.

**Reply by 0x56696B** [reply]
I'm using kitty. Neovim should behave the same as in any platform. Besides `dark-notify`, there isn't anything else platform specific. 
Also, I didn't have those issues several days ago.
I speculate it's some package I have, as rolling back the config to a prev version doesn't fix the issue.

**Reply by 0x56696B** [reply]
I figured out that disabling Treesitter's highlight with `TSDisable highlight` for the current buffer solves the issue for *some* of the highlights. 

Enabled:
<img width="516" alt="image" src="https://github.com/user-attachments/assets/829c916e-707f-46b8-89c4-3a0ddff37bf8">

Disabled:
<img width="501" alt="image" src="https://github.com/user-attachments/assets/c7683f33-aa8b-49c7-b8c6-d9533e91fa1d">

The `vim` keyword stops being highlighted, but the `if` statement persists.
Other `TSDisable` options provide no result.

**Reply by dpetka2001** [reply]
Try with a default LazyVim installation from scratch. I can't repro this behavior on my Linux Mint 21.3 with default LazyVim installation.

**Reply by 0x56696B** [reply]
Running the starter config in docker, I get the same issue.
Switching to `iTerm` (Mac OS's default one), also gives the same results

<img width="1274" alt="image" src="https://github.com/user-attachments/assets/ab454a6e-46e3-4287-8901-1afc86c67e7b">

<img width="1136" alt="image" src="https://github.com/user-attachments/assets/ad8305ac-4027-4e2a-9e09-d05bf2677fb2">

**Reply by dpetka2001** [reply]
Are you using tmux maybe?

**Reply by 0x56696B** [reply]
I am. Could that be the issue?

**Reply by 0x56696B** [reply, marked as answer]
Yep, detaching from `tmux` and running `nvim` now properly displays the colors

<img width="1368" alt="image" src="https://github.com/user-attachments/assets/d18dd7cc-defc-4f69-b70f-887b73ee1b56">

I'll go hunt the issue in `tmux` now. Thank you very much @dpetka2001 !

EDIT:

For all of those having the same issue and using `neovim` inside `tmux`, look into this setting:
```bash
set-option -sa terminal-features ',tmux-256color:RGB'
```

If you have it, changing to this fixed the issue for me.
```bash
set -g default-terminal "xterm-256color"
```

Again, huge thanks to @dpetka2001 for helping me solve the issue!
